api/bucket.go

```diff
@@ -1,19 +1,205 @@
 package api
 
 import (
-	"github.com/gin-gonic/gin"
-	"github.com/konveyor/tackle2-hub/model"
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"fmt"
 	"io"
+	"io/ioutil"
 	"net/http"
 	"os"
+	"path"
 	pathlib "path"
+	"path/filepath"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/konveyor/tackle2-hub/model"
 )
 
 //
 // BucketHandler provides bucket management.
 type BucketHandler struct {
 }
 
+// TODO: Validate Wildcard param to not allow access dirs outside of the bucket
+func (h *BucketHandler) serveBucketGet(ctx *gin.Context, owner *model.BucketOwner) {
+	if ctx.Request.Header.Get(Accept) == TarGzMimetype {
+		h.getDirArchive(ctx, path.Join(owner.Bucket, ctx.Param(Wildcard)))
+	} else {
+		h.content(ctx, owner)
+	}
+}
+
+func (h *BucketHandler) serveBucketUpload(ctx *gin.Context, owner *model.BucketOwner) {
+	if ctx.Request.Method == "PUT" {
```

Comment:
There is was a problem with mimetype, the file upload set the HTTP request to be ```multipart/form-data``` which wraps the uploaded tar.gz file. The API use POST&PUT for the same method (https://github.com/konveyor/tackle2-hub/blob/main/api/application.go#L39-L40), so it looked acceptable to use the PUT method in condition wheter a single file should be uploaded (as until now) or a file with tar.gz archive should be extracted to given path.

This approach can be discussed, but it looks that checking Content-Type of the PUT request for application/x-gzip doesn't seem to work.